Doc/library/socket.rst

```diff
@@ -959,6 +963,25 @@ The :mod:`socket` module also offers various network-related services:
    :const:`AF_INET6`), and is meant to be passed to the :meth:`socket.connect`
    method.
 
+   .. note::
+
+      If you intend to use results from :func:`!getaddrinfo` to create a socket
+      (rather than, for example, retrieve *canonname*),
+      consider limiting the results by *type* (e.g. :data:`SOCK_STREAM` or
+      :data:`SOCK_DGRAM`) and/or *proto* (e.g. :data:`IPPROTO_TCP` or
+      :data:`IPPROTO_UDP`) that your application can handle.
+
+      With default values of *family*, *type*, *proto* and/or *flags*,
+      many systems will return a sorted list of all matching addresses,
+      which should generally be tried in order until a connection succeeds
+      (possibly tried in parallel, for example, using a `Happy Eyeballs`_ algorithm).
+      In these cases, limiting the *type* and/or *proto* can help eliminate
+      unsuccessful or unusable connecton attempts.
+
+      Some systems will, however, only return a single address.
+      On these systems, limiting the *type* and/or *proto* helps ensure that
+      this address is usable.
```

Comment:
OK. With that in, I think it's best to add an extra paragraph to avoid putting too many thoughts in a single sentence.
```suggestion
      The behavior with default values of *family*, *type*, *proto*
      and *flags* is system-specific.
  
      Many systems (for example, most Linux configurations) will return a sorted
      list of all matching addresses.
      These addresses should generally be tried in order until a connection succeeds
      (possibly tried in parallel, for example, using a `Happy Eyeballs`_ algorithm).
      In these cases, limiting the *type* and/or *proto* can help eliminate
      unsuccessful or unusable connecton attempts.

      Some systems will, however, only return a single address.
      (For example, this was reported on Solaris and AIX configurations.)
      On these systems, limiting the *type* and/or *proto* helps ensure that
      this address is usable.
```

Reply:
👍